### ISSUE TITLE
Improve a11y of client-side routing

### ISSUE THREAD

**Issue by meganesu** [meganesu/personal-site-v2#34]
## Background

Right now, when keyboard users move through the site, their focus gets put at the top of the page every time they move to a new page. This means they have to spend a lot of time tabbing through the navigation over and over again.

## Acceptance Criteria

Based on [research by Marcy Sutton with Fable Labs](https://www.youtube.com/watch?v=Tr21FqQQv-U), a better pattern would be:

1. WHEN a user navigates to a new page via a link in the navbar\
THEN their focus moves to a control element (skip link) in the main header of the new page\
AND that control links back to the navbar.

1. WHEN a user navigates to a new blog post via a link in the /blog page\
THEN their focus moves to a control element (skip link) in the main header of the new page\
AND that control links back to the /blog page.

## Resources

* [YouTube video: Accessible client-side routing / Marcy Sutton and Samuel Proulx #ID24 2019](https://www.youtube.com/watch?v=Tr21FqQQv-U)
* [Gatsby blog post: What we learned from user testing of accessible client-side routing techniques with Fable Tech Labs](https://www.gatsbyjs.com/blog/2019-07-11-user-testing-accessible-client-routing/)

**Comment by meganesu** [comment]
- [ ] Build a RouteHeading component.
    - Should include a link to a targetId
- [ ] Use RouteHeading component on /blog page.
- [ ] Use RouteHeading component on /projects page.
- [ ] Use RouteHeading component on blog post template page.
- [ ] Use RouteHeading component on home page?

**Comment by meganesu** [comment]
Additional requirements:

* Screen reader should announce when user moves to a new page (existing functionality)

**Comment by meganesu** [comment]
Closed by #59
